kube/manifests/prometheus-gcs.yaml

```diff
@@ -18,16 +18,32 @@ spec:
         app: prometheus-gcs
         thanos-peer: "true"
     spec:
+## Commented out because Minikube has only one node, should be commented in for any production setup
+#      affinity:
+#        podAntiAffinity:
+#          requiredDuringSchedulingIgnoredDuringExecution:
+#          - labelSelector:
+#              matchExpressions:
+#              - key: app
+#                operator: In
+#                values:
+#                - prometheus
+#            topologyKey: kubernetes.io/hostname
       containers:
       - name: prometheus
         image: quay.io/prometheus/prometheus:v2.0.0
-        args:
-        - "--storage.tsdb.retention=24h"
-        - "--config.file=/etc/prometheus/prometheus.yml"
-        - "--storage.tsdb.path=/var/prometheus"
-        - "--storage.tsdb.min-block-duration=2h"
-        - "--storage.tsdb.max-block-duration=2h"
-        - "--web.enable-lifecycle"
+        command:
+        - "/bin/sh"
+        - "-c"
+        - >-
+            sed s/'@@HOSTNAME@@'/${HOSTNAME}/g /etc/prometheus/prometheus.yml.tmpl > /tmp/prometheus.yml
```

Comment:
Yea, this one way doing it. 

But actually IMO more preferable would be to use thanos sidecar envvar substitution logic:

1. Use here ` --config.file=/etc/prometheus-shared/prometheus.yaml` as it is.
2. Move this volume mount to thanos sidecar
```
         - name: config
           mountPath: /etc/prometheus
``` 
3. Add shared dir for both containers:
```
          - mountPath: /etc/prometheus-shared
            name: shared-config
```
3.  Add these flags to sidecar:
```
 --reloader.config-file=/etc/prometheus/prometheus.yaml.tmpl
 --reloader.config-envsubst-file=/etc/prometheus-shared/prometheus.yaml
```
4.  Use `replica: $(HOSTNAME)` for prometheus.yaml.tmpl replica external label.

This is how we do for prod setup. 
The essential advantage of this is not using "2-process within one docker hack" we do here (:

What do you think?